internal/kojiapi/openapi.yml

```diff
@@ -0,0 +1,168 @@
+openapi: 3.0.1
+info:
+  version: '1'
+  title: OSBuild Composer - Koji
+  description: Service to build and push images to Koji.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
+paths:
+  '/compose/{id}':
+    get:
+      summary: The status of a compose
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+            format: uuid
+            example: 123e4567-e89b-12d3-a456-426655440000
+          required: true
+          description: ID of compose status to get
+      description: 'Get the status of a running or finished compose. This includes whether or not it succeeded, and also meta information about the result.'
+      responses:
+        '200':
+          description: Compose status
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeStatus'
+        '400':
+          description: Invalid compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+        '404':
+          description: Unknown compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+  /compose:
+    post:
+      summary: Create compose
+      description: 'Create a new compose, potentially consisting of several images and upload each to koji.'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ComposeRequest'
+      responses:
+        '201':
+          description: Compose has started
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeResponse'
+        '400':
+          description: Invalid compose request
+          content:
+            text/plain:
+              schema:
+                type: string
+        '415':
+          description: The content type is not supported
+          content:
+            text/plain:
+              schema:
+                type: string
+components:
+  schemas:
+    ComposeStatus:
+      required:
+        - status
+        - image_statuses
+        - koji_task_id
+      properties:
+        status:
+          type: string
+          enum:
+            - success
+            - failure
+            - pending
+            - registering
+          example: success
+        image_statuses:
+          type: array
+          items:
+            $ref: '#/components/schemas/ImageStatus'
+        koji_task_id:
+          type: integer
+          example: 203143
+    ImageStatus:
+      required:
+        - status
+      properties:
+        status:
+          type: string
+          enum:
+            - success
+            - failure
+            - pending
+            - building
+            - uploading
```

Comment:
This seems also out of sync: `{'image_statuses': [{'status': 'FINISHED'}], 'koji_task_id': 0, 'status': 'FINISHED'}`, i.e. `FINISHED` not in the list.

Reply:
Fixed the implementation.